pf4j/src/main/java/ro/fortsoft/pf4j/DefaultPluginManager.java

```diff
@@ -658,6 +658,37 @@ protected PluginClasspath createPluginClasspath() {
     	return new PluginClasspath();
     }
 
+    protected PluginStatusProvider createPluginStatusProvider() {
+        return new PluginStatusProvider() {
```

Comment:
My idea is to move the default implementation in a `DefaultPluginStatusProvider`.